packages/truffle-compile/compilerSupplier/loadingStrategies/Docker.js

```diff
@@ -12,24 +12,19 @@ class Docker extends LoadingStrategy {
     const command =
       "docker run -i ethereum/solc:" + this.config.version + " --standard-json";
 
-    const versionRange = new VersionRange();
-    const commit = versionRange.getCommitFromVersion(versionString);
-
-    return versionRange
-      .getSolcByCommit(commit)
-      .then(solcjs => {
-        return {
+    return new Promise((resolve, reject) => {
+      try {
+        resolve({
           compile: options => String(execSync(command, { input: options })),
-          version: () => versionString,
-          importsParser: solcjs
-        };
-      })
-      .catch(error => {
+          version: () => versionString
+        });
+      } catch (error) {
         if (error.message === "No matching version found") {
-          throw this.errors("noVersion", versionString);
+          reject(this.errors("noVersion", versionString));
         }
-        throw new Error(error);
-      });
+        reject(new Error(error));
+      }
+    });
```

Comment:
I think we can get rid of the outer `new Promise` here.

All that's happening here is:

```javascript
try {
  return {
    compile: options => String(execSync(command, { input: options })),
    version: () => versionString
  };
} catch (error) {
  if (error.message === "No matching version found") {
    throw this.errors("noVersion", versionString);
  }
  throw new Error(error);
}
```

Because `load` is `async` already, it's going to return a Promise no matter what.

Reply:
Oh sweet! Completely missed that.